### PR TITLE
virtiofsd: Build inside a container

### PR DESCRIPTION
**Pull request by fidencio**
When moving to building the CI artefacts using the kata-deploy scripts, we've noticed that the build would fail on any machine where the tarball wasn't officially provided.

This happens as rust is missing from the 1st layer container.  However, it's a very common practice to leave the 1st layer container with the minimum possible dependencies and install whatever is needed for building a specific component in a 2nd layer container, which virtiofsd never had.

In this commit we introduce the second layer containers (yes, comtainers), one for building virtiofsd using musl, and one for building virtiofsd using glibc.  The reason for taking this approach was to actually simplify the scripts and avoid building the dependencies (libseccomp, libcap-ng) using musl libc.

Fixes: #5425

Signed-off-by: Fabiano Fidêncio <fabiano.fidencio@intel.com>

### CODE REVIEW

**Comment by fidencio** [comment]
@BbolroC @jongwu @Amulyam24, I'd like you to test this by doing a:
`make virtiofsd-tarball` from the top directory of the Kata Containers repo.

Please, let me know if any failure happens.

**Comment by BbolroC** [comment]
Hi, no errors on s390x
```
$ git clone -b topic/build-virtiofsd-in-a-2nd-layer-container https://github.com/fidencio/kata-containers.git
Cloning into 'kata-containers'...                                                                                                                                                                                                            
$ cd kata-containers/                                                                                     
$ make virtiofsd-tarball
... build logs ...
INFO: Install static virtiofsd
./
./opt/
./opt/kata/
./opt/kata/libexec/
./opt/kata/libexec/virtiofsd
drwxr-xr-x jenkins/jenkins   0 2022-10-14 08:39 ./
drwxr-xr-x jenkins/jenkins   0 2022-10-14 08:39 ./opt/
drwxr-xr-x jenkins/jenkins   0 2022-10-14 08:39 ./opt/kata/
drwxr-xr-x jenkins/jenkins   0 2022-10-14 08:39 ./opt/kata/libexec/
-rwxr--r-- root/root   3397048 2022-10-14 08:39 ./opt/kata/libexec/virtiofsd
make[1]: Leaving directory '/home/jenkins/test/kata-containers'
$ echo $?
0
```

File: tools/packaging/static-build/virtiofsd/build-static-virtiofsd.sh [hunk 1]
```diff
@@ -16,10 +16,13 @@ script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 
 source "${script_dir}/../../scripts/lib.sh"
 
+virtiofsd_repo="${virtiofsd_repo:-}"
 virtiofsd_version="${virtiofsd_version:-}"
+virtiofsd_zip="${virtiofsd_zip:-}"
 
-[ -n "$virtiofsd_version" ] || virtiofsd_version=$(get_from_kata_deps "externals.virtiofsd.version")
+[ -n "$virtiofsd_repo" ] || die "failed to get virtiofsd repo"
 [ -n "$virtiofsd_version" ] || die "failed to get virtiofsd version"
```
Comment (stevenhorsman) [inline tools/packaging/static-build/virtiofsd/build-static-virtiofsd.sh:24]: minor nit and free feel to ignore it and it wasn't introduced in this PR, but is it worth considering using parameter expansion on `$virtiofsd_repo` and `$virtiofsd_version `? I got burned by this in the past so try and be very liberal with using braces everywhere now!

**Review by stevenhorsman — APPROVED** [review]
LGTM and helps unify and tidy up the code. Thanks!

**Review by wainersm — APPROVED** [review]
I talked with @fidencio about the fact that we are using the latest version of rust, whereas we better use a pinned version. This is something we will address on the next bump of virtiofsd.

Thanks @fidencio !

**Review by amshinde — APPROVED** [review]
lgtm @fidencio.
A small nit, instead of saying second layer containers, which was a bit confusing to me considering containers use layers, I would reword that as nested containers.
